clients/pkg/logentry/stages/metrics.go

```diff
@@ -321,3 +321,21 @@ func getFloat(unk interface{}) (float64, error) {
 		return math.NaN(), fmt.Errorf("can't convert %v to float64", unk)
 	}
 }
+
+// getFloatFromString converts string into float64
+// Two types of string formats are supported:
+//   * strings that represent floating point numbers, e.g., "0.804"
+//   * duration format strings, e.g., "0.5ms", "10d".
+//     Values in this format are converted as a floating point number of seconds.
+//     E.g., "0.5ms" is converted to 0.0005
+func getFloatFromString(str string) (float64, error) {
+	dur, err := strconv.ParseFloat(str, 64)
+	if err != nil {
+		dur, err := time.ParseDuration(str)
```

Comment:
@dannykopping you suggested using `ParseDuration` from github.com/prometheus/common/model instead of the one from time package. They support different time units:
* prometheus: supports bigger time units "y", "w", "d", "h", "m", "s", "ms"
https://github.com/grafana/loki/blob/main/vendor/github.com/prometheus/common/model/time.go#L186
* time: supports smaller time units "ns", "us", "ms", "s", "m", "h"

I would say that the one from time package has the most suitable set for this use case. Wdyt?

Reply:
I tend to agree; I think the smaller units are probably more valuable here. Good call